src/suit_directive.c

```diff
@@ -476,7 +476,11 @@ int suit_directive_process_dependency(struct suit_processor_state *state, struct
 	manifest_state = &state->manifest_stack[state->manifest_stack_height - 1];
 
 	if (seq_exec_state->retval != SUIT_SUCCESS) {
-		retval = seq_exec_state->retval;
+		/** Failure of processing a dependency should not lead to a hard failure.
+		 *  The higher level manifest might need to implement a fallback mechanism
+		 *  if memory was damaged by the dependency.
+		 */
+		retval = SUIT_FAIL_CONDITION;
```

Comment:
Would not that be safer to do so in line 558?

Reply:
Ok, changed